dnf/module/module_base.py

```diff
@@ -140,31 +140,139 @@ def install(self, module_specs, strict=True):
+                                               module_depsolv_errors=solver_errors)
 
-        for pkg_name, set_specs in install_dict.items():
-            query = install_base_query.filter(name=pkg_name)
-            if not query:
-                # package can also be non-modular or part of another stream
-                query = base_no_source_query.filter(name=pkg_name)
-                if not query:
-                    for spec in set_specs:
-                        logger.error(_("Unable to resolve argument {}").format(spec))
-                    logger.error(_("No match for package {}").format(pkg_name))
-                    error_specs.extend(set_specs)
-                    continue
-            self.base._goal.group_members.add(pkg_name)
+    def switch(self, module_specs, strict=True):
+        # :api
+        no_match_specs, error_specs, module_dicts = self.__resolve_specs_enable(module_specs)
+        # collect name of artifacts from new modules for distrosync
+        new_artefacts_names = set()
+        # collect name of artifacts from active modules for distrosync before sack update
+        active_artefacts_names = set()
+        src_arches = {"nosrc", "src"}
+        for spec, (nsvcap, moduledict) in module_dicts.items():
+            for name in moduledict.keys():
+                for module in self.base._moduleContainer.query(name, "", "", "", ""):
+                    if self.base._moduleContainer.isModuleActive(module):
+                        for artifact in module.getArtifacts():
+                            arch = artifact.rsplit(".", 1)[1]
+                            if arch in src_arches:
+                                continue
+                            pkg_name = artifact.rsplit("-", 2)[0]
+                            active_artefacts_names.add(pkg_name)
+
+        solver_errors = self._update_sack()
+
+        dependency_error_spec = self._enable_dependencies(module_dicts)
+        if dependency_error_spec:
+            error_specs.extend(dependency_error_spec)
+
+        # <package_name, set_of_spec>
+        fail_safe_repo = hawkey.MODULE_FAIL_SAFE_REPO_NAME
+        install_dict = {}
+        install_set_artefacts = set()
+        fail_safe_repo_used = False
+
+        # list of name: [profiles] for module profiles being removed
+        removed_profiles = self.base._moduleContainer.getRemovedProfiles()
+
+        for spec, (nsvcap, moduledict) in module_dicts.items():
+            for name, streamdict in moduledict.items():
+                for stream, module_list in streamdict.items():
+                    install_module_list = [x for x in module_list
+                                           if self.base._moduleContainer.isModuleActive(x.getId())]
+                    if not install_module_list:
+                        logger.error(_("All matches for argument '{0}' in module '{1}:{2}' are not "
+                                       "active").format(spec, name, stream))
```

Comment:
The english isn't correct here, how about this?
> No active matches for argument '{0}' in module '{1}:{2}'

Reply:
Nice :+1: